presto-product-tests/src/main/java/io/prestosql/tests/hive/acid/TestFullAcidTableRead.java

```diff
@@ -13,36 +13,127 @@
+
+            String selectFromOnePartitionsSql = "SELECT col, fcol FROM " + tableName + " ORDER BY col";
+            QueryResult onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsOnly(row(21, 1));
+
+            onHive().executeQuery("INSERT INTO TABLE " + tableName + getHivePartitionString(type.isPartitioned()) + " VALUES (22, 2)");
+            onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsExactly(row(21, 1), row(22, 2));
+
+            // test filtering
+            onePartitionQueryResult = query("SELECT col, fcol FROM " + tableName + " WHERE fcol = 1 ORDER BY col");
+            assertThat(onePartitionQueryResult).containsOnly(row(21, 1));
+
+            // delete a row
+            onHive().executeQuery(
+                    "DELETE FROM " + tableName + " where fcol=2");
+            onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsOnly(row(21, 1));
+
+            // update the existing row
+            onHive().executeQuery(
+                    "UPDATE " + tableName + " set col = 23 " + getPrestoPartitionPredicate(type.isPartitioned(), "fcol = 1"));
+            onePartitionQueryResult = query(selectFromOnePartitionsSql);
+            assertThat(onePartitionQueryResult).containsOnly(row(23, 1));
         }
         finally {
             onHive().executeQuery("DROP TABLE " + tableName);
         }
     }
+
+    private static String getHivePartitionString(boolean isPartitioned)
+    {
+        if (!isPartitioned) {
+            return "";
+        }
+
+        return " PARTITION (part_col=2) ";
+    }
+
+    private static String getPrestoPartitionPredicate(boolean isPartitioned, String columnPredicate)
+    {
+        String predicate = " WHERE " + columnPredicate;
+        if (isPartitioned) {
+            predicate += " AND part_col = 2 ";
+        }
+        return predicate;
+    }
+
+    private static void createTable(String tableName, boolean isPartitioned)
+    {
+        StringBuilder builder = new StringBuilder()
+                .append("CREATE TABLE IF NOT EXISTS ")
+                .append(tableName)
+                .append(" (col INT,")
+                .append("fcol INT) ");
+        if (isPartitioned) {
+            builder.append("PARTITIONED BY (part_col INT) ");
+        }
+
+        builder.append("STORED AS ORC ")
+                .append("TBLPROPERTIES ('transactional'='true') ");
+
+        onHive().executeQuery(builder.toString());
+    }
+
+    @DataProvider
+    public Object[][] fullAcidTableTypes()
+    {
+        return new Object[][] {
+                {FullAcidTableType.UNPARTITIONED},
+                {FullAcidTableType.PARTITIONED}
+        };
+    }
+
+    private enum FullAcidTableType
```

Comment:
Use boolean, just like in `io.prestosql.tests.hive.acid.TestInsertOnlyAcidTableRead#isTablePartitioned`

Reply:
In fact, we should unify the two test classes now. Can be a follow-up.